### PR TITLE
[SourceKit] Introduce "source.lang.swift.expr.tuple" (SR-5977)

### PR DESCRIPTION
**Pull request by marcelofabri**
This PR introduces `source.lang.swift.expr.tuple` to represent tuple expressions, similar to `source.lang.swift.expr.array` and `source.lang.swift.expr.dictionary`.

Currently this only works for tuple expressions (i.e. it won't report a tuple in types as `let x: (Y, Z)`). However, probably this is the expected behavior for expression kinds anyway.

This also doesn't provide any extra information for named tuples, but I think this can be done later.

Resolves [SR-5977](https://bugs.swift.org/browse/SR-5977).

// cc @nkcsgexi

### CODE REVIEW

File: lib/IDE/SyntaxModel.cpp [hunk 1]
```diff
@@ -507,10 +507,29 @@ std::pair<bool, Expr *> ModelASTWalker::walkToExprPre(Expr *E) {
...
+      }
+    } else {
+      SyntaxStructureNode SN;
+      SN.Kind = SyntaxStructureKind::TupleExpression;
+      SN.Range = charSourceRangeFromSourceRange(SM, Tup->getSourceRange());
+      SN.BodyRange = innerCharSourceRangeFromSourceRange(SM,
+                                                         Tup->getSourceRange());
+
+      for (auto *Elem : Tup->getElements()) {
+        if (auto *TupleE = dyn_cast<TupleExpr>(Elem)) {
```
Comment (nkcsgexi) [inline lib/IDE/SyntaxModel.cpp]: Why do we only care about the second level as a tuple and why do we flatten it?

Comment (marcelofabri) [reply lib/IDE/SyntaxModel.cpp]: Because I copied and pasted from somewhere else 😅 

Fixed!

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by marcelofabri** [comment]
The CI failure seems unrelated.

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci Please clean smoke test Linux platform

**Pull request merged**